File: MATLAB/src/utils/utils/interp_to.m
function Yt = interp_to(t_src, Y_src, t_tgt)
%INTERP_TO 1-D interp of vectors or NxM arrays along rows over time.
%   YT = INTERP_TO(T_SRC, Y_SRC, T_TGT) interpolates the data Y_SRC sampled
%   at times T_SRC to the target times T_TGT. Linear interpolation with
%   extrapolation is used. T_SRC and T_TGT are column vectors in seconds.
%   Y_SRC can be a vector or an N-by-M matrix where N = length(T_SRC).

% Ensure column vectors of type double
t_src = double(t_src(:));
t_tgt = double(t_tgt(:));

% Convert Y_src to 2-D matrix with rows corresponding to time samples
if isvector(Y_src)
    Y_src = Y_src(:);
end

% Preallocate output
Yt = zeros(numel(t_tgt), size(Y_src,2));

% Interpolate each column individually
for j = 1:size(Y_src,2)
    Yt(:,j) = interp1(t_src, Y_src(:,j), t_tgt, 'linear', 'extrap');
end
end

File: PYTHON/src/utils/interp_to.py
"""Time-series interpolation helper matching the MATLAB version."""
from __future__ import annotations
import numpy as np

def interp_to(t_src: np.ndarray, y_src: np.ndarray, t_tgt: np.ndarray) -> np.ndarray:
    """Interpolate *y_src* sampled at *t_src* to the times *t_tgt*.

    Linear interpolation with extrapolation is used to mirror the MATLAB
    helper.
    """
    t_src = np.asarray(t_src, dtype=float).reshape(-1)
    t_tgt = np.asarray(t_tgt, dtype=float).reshape(-1)
    y_src = np.asarray(y_src, dtype=float)
    if y_src.ndim == 1:
        y_src = y_src.reshape(-1, 1)
    out = np.empty((t_tgt.size, y_src.shape[1]), dtype=float)
    for j in range(y_src.shape[1]):
        out[:, j] = np.interp(t_tgt, t_src, y_src[:, j], left=None, right=None)
    return out
